infra/gcp_service_account/main.tf

```diff
@@ -1,6 +1,7 @@
 data "google_project" "project" {}
 
 resource "google_service_account" "dss" {
+  count = "${length(var.DSS_GCP_SERVICE_ACCOUNT_NAME) > 0 ? 1 : 0}"
```

Comment:
In Orange, we are generating all `.tf.json` files from pure Python templates. I found that pattern to be very useful because I can use comprehensions and conditionals and, well, Python as a Turing-complete language.

Here is a template with a list comprehension:

https://github.com/DataBiosphere/azul/blob/develop/terraform/api_gateway_domain_name.tf.json.template.py#L74

Reply:
Part of the aspirations of TF seems to be to provide a human readable record of deployed infrastructure that is relatively easy to understand. I would argue that this is defeated by auto-generated TF files, extra logic/conditionals, and even TF's weird interpolation syntax. But ymmv.

Maybe the solution is to create separate GCP service accounts for `integration` and `staging`. Then these conditionals will not be necessary, and I'll be less hypocritical.